scripts/quests/jeuno/Apocalypse_Nigh.lua

```diff
@@ -0,0 +1,290 @@
+                        return quest:progressEvent(4)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [4] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 2)
+                    player:setPos(-419.995, 0, 248.483, 191, 35) -- Puts player in the Garden of Ru'Hmet
+                end,
+            },
+        },
+
+        [xi.zone.EMPYREAL_PARADOX] =
+        {
+            ['Transcendental_Radiance'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'Prog') == 2 then
+                        return quest:progressEvent(4)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [4] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 3)
+                end,
+
+                [32001] = function(player, csid, option, npc)
+                    if quest:getVar(player, 'Prog') == 3 then
+                        quest:setVar(player, 'Prog', 4)
+                        player:startEvent(7)
+                    end
+                end,
+
+                [7] = function(player, csid, option, npc)
+                    player:setPos(-0.0745, -10, -465.1132, 63, 33)
+                end,
+            }
+        },
+
+        [xi.zone.LOWER_JEUNO] =
+        {
+            ['Aldo'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'Prog') == 4 and player:getRank(player:getNation()) > 5 then
+                        return quest:progressEvent(10057)
+                    elseif quest:getVar(player, 'Prog') == 5 then
+                        return quest:event(10058)
+                    end
+                end,
+            },
+
+            ['Sattal-Mansal'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'Prog') == 5 then
+                        return quest:progressEvent(10061)
+                    end
+                end,
+            },
+            ['Yin_Pocanakhu'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'Prog') == 5 then
+                        return quest:progressEvent(10060)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [10057] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 5)
+                    quest:setVar(player, 'Wait', getVanaMidnight())
+                end
+            },
+        },
+
+        [xi.zone.NORG] =
+        {
+            ['Gilgamesh'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'Prog') == 5 and os.time() >= quest:getVar(player, 'Wait') then
```

Comment:
Instead of using os.time() for a game day wait, would suggest looking at VanadielUniqueDay() and its usage throughout missions and quests